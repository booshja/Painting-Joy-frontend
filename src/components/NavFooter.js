// dependencies
import React from "react";
import styled from "styled-components";
// components
import { Link } from "react-router-dom";

const StyledNav = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 125px;
`;

const StyledLink = styled(Link)`
    font-family: "News Cycle", sans-serif;
    font-size: 1.2rem;
    letter-spacing: 1px;
    color: #ffffff;
`;

const NavFooter = () => {
    return (
        <StyledNav>
            <StyledLink to="/">Home</StyledLink>
            <StyledLink to="/murals">Murals</StyledLink>
            {/* <StyledLink to="/more-art">More Art</StyledLink> */}
            <StyledLink to="/contact">Contact</StyledLink>
            <StyledLink to="/store">Store</StyledLink>
        </StyledNav>
    );
};

export default NavFooter;
