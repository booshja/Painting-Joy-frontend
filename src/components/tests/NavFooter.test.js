import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NavFooter from "../NavFooter";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <NavFooter />
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <NavFooter />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText } = render(
        <BrowserRouter>
            <NavFooter />
        </BrowserRouter>
    );

    getByText("Home");
    getByText("Murals");
    getByText("Contact");
    getByText("Store");
});
