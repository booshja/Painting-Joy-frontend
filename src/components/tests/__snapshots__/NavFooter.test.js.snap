// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM eEVwcp"
  >
    <a
      class="sc-gtsrHT ickbIz"
      href="/"
    >
      Home
    </a>
    <a
      class="sc-gtsrHT ickbIz"
      href="/murals"
    >
      Murals
    </a>
    <a
      class="sc-gtsrHT ickbIz"
      href="/contact"
    >
      Contact
    </a>
    <a
      class="sc-gtsrHT ickbIz"
      href="/store"
    >
      Store
    </a>
  </div>
</DocumentFragment>
`;
